core.lua

```diff
@@ -89,14 +89,51 @@ function Simulationcraft:UpdateMinimapButton()
   end
 end
 
+local function splitLinks(links)
+  local separatedLinks = {}
+  while true do
+    local _,firstLinkEnd = strfind(links, '|r')
+    if firstLinkEnd == strlen(links) then -- link only contains one link.
+      separatedLinks[#separatedLinks + 1] = links
+      break
+    end
+
+    separatedLinks[#separatedLinks + 1] = strsub(links, 0, firstLinkEnd)
+    links = strsub(links, firstLinkEnd + 1)
+  end
+  return separatedLinks
+end
+
 function Simulationcraft:HandleChatCommand(input)
   local args = {strsplit(' ', input)}
 
   local debugOutput = false
   local noBags = false
+  local links = {}
```

Comment:
With the above function, the link detection code can be way simpler.

Just need to have

```
local links = splitLinks(input)
```

And it works out. The rest of the arg parsing will not match any of the link fragments but still catch any of the keywords.

Reply:
Done!